### ISSUE TITLE
Upstream `pkgs/servers/mail/opensmtpd/proc_path.diff`

### ISSUE THREAD

**Issue by Ekleog** [NixOS/nixpkgs#54045]
@gebner @obadz I'm not sure the license for the patch in nixpkgs is compatible with the license of OpenSMTPD, so could you confirm you are OK with upstreaming https://github.com/NixOS/nixpkgs/blob/master/pkgs/servers/mail/opensmtpd/proc_path.diff to OpenSMTPD? I don't know yet whether it'd be accepted but it's always worth a try, and if I read correctly the git log you're the only ones with non-trivial contributions to it.

**Comment by gebner** [comment]
Yes, I think I did the original patch: https://github.com/NixOS/nixpkgs/pull/9955#issuecomment-143408086

I'd be very happy to have the patch upstreamed, feel free to use it under OpenSMTPD's ISC license.

**Comment by obadz** [comment]
Had already tried: https://github.com/OpenSMTPD/OpenSMTPD/pull/697

What's the license issue?

**Comment by Ekleog** [comment]
The license issue was just the current patch being under NixOS's MIT-with-exception and OpenSMTPD being under some weird mix of BSD licenses ; so it was basically me asking “is it OK with you (who wrote the patch) if I try to upstream?”.

But the link you sent appears to mean the patch won't be upstream-able as-is, and thus we'll need some more work to make it into something properly configurable… I'll be adding a link around the `proc_path.diff` pointing to this issue so as to point to why this isn't upstream (yet) :)

**Comment by obadz** [comment]
@Ekleog, I don't remember writing the patch but to the extent I contributed to it, feel free to use my contributions under whatever license you see fit.

**Comment by flokli** [comment]
This still doesn't seem to have ended upstream, let's reopen it for now.
